stdlib/src/utils/string_slice.mojo

```diff
@@ -987,6 +1015,77 @@ trait Stringlike:
         ...
 
 
+fn _to_string_list[
+    T: CollectionElement, //,
+    len_fn: fn (T) -> Int,
+    unsafe_ptr_fn: fn (T) -> UnsafePointer[Byte],
+](items: List[T]) -> List[String]:
+    i_len = len(items)
+    i_ptr = items.unsafe_ptr()
+    out_ptr = UnsafePointer[String].alloc(i_len)
+
+    for i in range(i_len):
+        og_len = len_fn(i_ptr[i])
+        f_len = og_len + 1  # null terminator
+        p = UnsafePointer[Byte].alloc(f_len)
+        og_ptr = unsafe_ptr_fn(i_ptr[i])
+        memcpy(p, og_ptr, og_len)
+        p[og_len] = 0  # null terminator
+        buf = String._buffer_type(unsafe_pointer=p, size=f_len, capacity=f_len)
+        (out_ptr + i).init_pointee_move(String(buf^))
+    return List[String](unsafe_pointer=out_ptr, size=i_len, capacity=i_len)
+
+
+@always_inline
+fn to_string_list[
```

Comment:
**Question** Even though `String` and `StringLiteral` are using this, do we want this to be a public API?  I'd prefer to keep it private for now.

Reply:
Ok, I'll change it to private, though I think this will be useful since nobody is going to bother writing such a helper using `UnsafePointer` and doing manual `(out_ptr + i).init_pointee_move(String(buf^))` and `memcpy`